packages/mobile/src/store/init/setupCrypto/CryptoDelegator.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  CryptoServicePayload,
+  CryptoServiceResponse,
+  SocketActionTypes
+} from '@quiet/state-manager'
+import { Socket } from 'socket.io-client'
+import { NodeEnv } from '../../../utils/const/NodeEnv.enum'
+
+export interface ArrayBufferViewWithPromise extends ArrayBufferView {
+  _promise?: Promise<ArrayBufferView>
+}
+
+const SUBTLE_METHODS = [
+  'encrypt',
+  'decrypt',
+  'sign',
+  'verify',
+  'digest',
+  'generateKey',
+  'deriveKey',
+  'deriveBits',
+  'importKey',
+  'exportKey',
+  'wrapKey',
+  'unwrapKey'
+]
+
+export class CryptoDelegator {
+  socket: Socket
+
+  calls: {
+    [id: string]: {
+      resolvePromise: (value: any) => void
+    }
+  }
+
+  constructor(socket: Socket) {
+    this.socket = socket
+  }
+
+  public get subtle(): SubtleCrypto {
+    const subtle = {}
+    for (const m of SUBTLE_METHODS) {
+      subtle[m] = async (...args) => {
+        const call = await this.call(`subtle.${m}`, args)
+        return call
+      }
+    }
+    return subtle as SubtleCrypto
+  }
+
+  private call = async (method: string, args: any[]): Promise<any> => {
+    const id = CryptoDelegator.uuid()
+    // store this promise, so we can resolve it when we get a value
+    // back from the crypto service
+    let resolvePromise
+    const promise = new Promise(resolve => {
+      resolvePromise = resolve
```

Comment:
reject is also important; you've ignore posibility that backend will return with error